x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/processor_form/processors/common_fields/common_processor_fields.tsx

```diff
@@ -18,7 +18,7 @@ import {
 import { TextEditor } from '../../field_components';
 import { to, from, EDITOR_PX_HEIGHT } from '../shared';
 
-const ignoreFailureConfig: FieldConfig = {
+const ignoreFailureConfig: FieldConfig<any, any> = {
```

Comment:
I know we discussed some of this offline, but seeing it now I am wondering if `FieldConfig` could be typed as:

```ts
interface FieldConfig<ValueType = unknown, T extends FormData = any> {}
```

Then when it is used on individual fields we can write:

```ts
const ignoreFailureConfig: FieldConfig<boolean> = {
 ...
}
```

I'm not certain how often we will set a type for FieldData when using this type directly, I can see it provides types for the validator, but perhaps the helpers/validators we provide cover a large number of cases already?

Reply:
Yes it makes total sense, I will update it 👍